### PR TITLE
Support specifying TiSpark version in TiUP cluster

### PR DESCRIPTION
**Pull request by shiyuhang0**
### What problem does this PR solve? 
Support specifying tispark version.
This pr should not be merged before upload spark & tispark to mirror 

### What is changed and how it works?
- Support specifying TiSpark version
   - Add spark_version, tispark_version and scala_version in TiSparkMasterSpec
   - valid version: spark_version and tispark_version should be set together or not be set together
   - Transfer spark_version, tispark_version, and scala_version to TiSparkInstance(both master and worker)
   - Get version info and download spark & tispark
   - Choose the TiSpark jar  to deploy
   - Get version info from mergedtopo when scale-out
- Generate different configs for different tispark_version
- Skip  TiSpark when upgrade

### Check List 

Tests 

Manual test (add detailed scripts or steps below)
 - spark_version 3.0.3 tispark version 3.0.0 scala version 2.12 (success)
 - spark_version 3.0.3 tispark version 3.0.0 (success)
 - spark_version 2.4.3 tispark version 2.4.1 (success)
 - no paratemer (success)
 - spark_version 3.0.3 tispark version 3.0.0 scala version 2.11 (fail because no jar found)
 - spark_version 3.0.3 (fail because tispark version not set)

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change

Related changes

 - Need to update the documentation

Release notes:

```release-note
NONE
```

### CODE REVIEW

**Comment by shiyuhang0** [comment]
@AstroProfundis @nexustar PTAL

**Comment by shiyuhang0** [comment]
@AstroProfundis @nexustar PTAL

File: pkg/cluster/manager/deploy.go [hunk 1]
```diff
@@ -275,15 +271,16 @@ func (m *Manager) Deploy(
...
-				if sparkVer, _, iterErr = env.V1Repository().WithOptions(repository.Options{
-					GOOS:   inst.OS(),
-					GOARCH: inst.Arch(),
-				}).LatestStableVersion(spec.ComponentSpark, false); iterErr != nil {
-					return
+				// Get spark version and TiSpark version from master or worker
+				var sparkVersion, tiSparkVersion, scalaVersion string
+				if len(topo.(*spec.Specification).TiSparkMasters) > 0 {
+					sparkVersion, tiSparkVersion, scalaVersion = topo.(*spec.Specification).TiSparkMasters[0].GetVersion()
+				} else {
```
Comment (AstroProfundis) [inline pkg/cluster/manager/deploy.go:278]: Is this possible? I think we have a check in spec that does not allow master-less topology

Comment (shiyuhang0) [reply pkg/cluster/manager/deploy.go:278]: yes, it is impossible in theory. but do you think it is necessary to do this judge for unexpected situations?

**Pull request closed without merging**